router/pkg/config/config.schema.json

```diff
@@ -1362,7 +1362,42 @@
     "introspection_enabled": {
       "type": "boolean",
       "description": "Enable the GraphQL introspection. The GraphQL introspection allows you to query the schema of the GraphQL API. The default value is true. If the value is false, the GraphQL introspection is disabled. In production, it is recommended to disable the introspection.",
-      "default": true
+      "default": true,
+      "deprecated": true,
+      "deprecationMessage": "introspection_enabled is deprecated. Please use the introspection.enabled configuration instead."
+    },
+    "introspection": {
+      "type": "object",
+      "description": "",
+      "additionalProperties": false,
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "",
+          "default": true
+        },
+        "authentication_mode": {
```

Comment:
I find these variations confusing. We either check for auth or not and allow optionally a token to validate. We should always avoid creating configurations that depend on each other on the same level. Have you thought about simplifying it?

```yaml
introspection:
  enabled: true
  skip_auth: false
  token: ".."
```

Reply:
Funnily enough, I suggested the opposite of this to make it the way it is now because the interactions between `token` and `skip_auth` are not clear in this way.

Perhaps:


```yaml
introspection:
  enabled: true
  skip_auth: true <- use no auth
  token_auth: "..." <- use token auth
```

Where `skip_auth` and `token_auth` are mutually exclusive enforced by schema?

Reply:
Got it. How about?

```yaml
introspection:
  enabled: true
  secret: "..." # empty by default
  
authentication:
  ignore_introspection: true # false by default
```

Reply:
Yeah its hard to let the user know intuitively what the config parameters are meaning and how they interact without pointing to the docs. What about this approach:

```yaml
introspection:
  enabled: true
  skip_auth: false
  skip_auth_token: ".."
```

Thats @StarpTech ´s approach with a renamed `token` parameter. The downside is requiring a token to skip auth is still some sort of auth. So it reads a bit confusing ("skip auth by using auth"). But I think to the user it brings across the meaning. At least as long as they read it as `-->introspection<--.skip_auth_token`. 

Me personally I am not too invested and I think both approaches will work, just have to settle for one.

Reply:
I agree with your `"skip auth by using auth"`. I'd use the `authentication ` section to let the user skip auth for introspection. That's why we have the section for. Optionally, a User can then define a secret / token to protect introspection calls. IMO this is the least confusing approach.